gsoc/GSOC_2017_Application_proposal.md

```diff
@@ -0,0 +1,70 @@
+
+**Describe the need your project fills:**  
+The broader goal of the project is to build an assistant system to reduce the work of analysts. This kind of reduction means: 1) Deal with the data computation with preprocessing and reduce all unnecessary extra work so that the system can provide the useful information as much as possible to the analysts. 2) Detect the changes of the target websites and identify the significance of these website changes accurately and efficiently.
+
+**Describe how your project meets this need:**  
+To reach the goal above, the project overview has already shown a very clear picture of the use case and the corresponding definitions, architecture etc. But there are still some places that need to be considered in depth. Here are two aspects which Dr. Hucka talked to me. I think it is really worth digging:
+
+1>. What is the optimal format of the input?
+2>. What is the purpose when the analysts analyze the diffs?
+
+For the first problem (1>), one possible way to monitor the website changes is Versionista. It seems the input data will have the JSON format. It can give a specific description of the changes but it is not easy to be used for analysis. Besides, I checked the diff function of the Unix or Linux (e.g., http://www.computerhope.com/unix/udiff.htm). It can give you a similar result but we can see this kind of result still need to be refined. 
+
+For the second problem (2>), it closely related to how does the “significant change” define. The target of monitoring depends on the analysts, but I think the significant change means there is at least one index exceeds the expectation of this analyst a lot. So it should have a multi-perspective significant change definition and a overall significant change. But we need to think about what is the "index" we need, how many and what are this kind of indexes are the optimal choices etc. 
+
+Currently I have a a N-component definition idea of the input data to solve the problem. For example, it includes how many changes happens during a specific period of time, how may JSON blobs are changed, what is the time interval when the changes happen, what is the frequency of each change during the monitoring time period etc. All observed data can be reformulated to be the same structure input data. And each component corresponds a weight. Different analyst may have different opinions of the weights distribution. But this kind of input data structure can build a quantized framework of the input data to reflect the changes. The analysts can have their own ways to think about which part is the required significant change. And the good thing is that this kind of data structure can help a unifed way to compare different websites significant changes by using the same weights. 
+
+**Milestones/Timeline:**  
+Week1: Processing and observing data, give a statistical analysis of the samples and formulate the input data’s structure.
```

Comment:
The formatting here did not turn out very well. A little bit of tweaking to use Markdown lists or tables would make this a little bit more readable.